src/algebra/group_power/basic.lean

```diff
@@ -623,6 +623,19 @@ pow_bit0_pos h 1
 
 end linear_ordered_ring
 
+/- this lemma would work for `ordered_integral_domain`, if that typeclass existed -/
```

Comment:
I think you can weaken the requirements by changing line 406 to
```
eq_or_eq_neg_of_pow_two_eq_pow_two [comm_ring R] [no_zero_divisors R]
```
which eliminates the need for `integral_domain`

Reply:
I switched to `[linear_ordered_comm_ring R] [no_zero_divisors R] `.  Doesn't seem to work without the linearity of the order, probably some quirk of the structure hierarchy.